pkg/azure/blob/credentials.go

```diff
@@ -26,7 +47,11 @@ func GetCredentials(cfg azureconfig.Config, l hclog.Logger) (CredentialSet, erro
 
 	connString := os.Getenv(credsEnv)
 	if connString == "" {
-		return CredentialSet{}, errors.Errorf("environment variable %s containing the azure storage connection string was not set\n%#v", credsEnv, cfg)
+		cred, err := GetCredentialsFromCli(cfg, l)
+		if err != nil {
+			return CredentialSet{}, errors.Errorf("environment variable %s containing the azure storage connection string was not set: %v\n%#v", credsEnv, err, cfg)
```

Comment:
I'm confused with this error... I was thinking if CLI credentials were being used in tandem with the provided account and resource group values, the environment variable AZURE_STORAGE_CONNECTION_STRING wouldn't be needed?

Reply:
What I was trying to achieve here was saying that AZURE_STORAGE_CONNECTION_STRING so we then tried to use Azure CLI and that had an issue too, if its not clear I can remove the outer error?

Reply:
Oh I see. Hmm, maybe we update the outer error to something like "unable to get credentials from that Azure CLI...`?